src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs

```diff
@@ -0,0 +1,293 @@
+﻿#nullable enable
+
+using System;
+using System.Composition;
+using System.IdentityModel.Tokens.Jwt;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using DevToys.Helpers;
+using DevToys.Helpers.JsonYaml;
+using DevToys.Models;
+using DevToys.Models.JwtDecoderEncoder;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.UI.Xaml.Controls;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.OpenSsl;
+
+namespace DevToys.ViewModels.Tools.EncodersDecoders.JwtDecoderEncoder
+{
+    [Export(typeof(JwtDecoder))]
+    [Shared]
+    public class JwtDecoder
+    {
+        private const string PublicKeyStart = "-----BEGIN PUBLIC KEY-----";
+        private const string PublicKeyEnd = "-----END PUBLIC KEY-----";
+        private Action<TokenResultErrorEventArgs>? _decodingErrorCallBack;
+        private JwtDecoderEncoderStrings _localizedStrings => LanguageManager.Instance.JwtDecoderEncoder;
+
+        public TokenResult? DecodeToken(
+                DecoderParameters decodeParameters,
+                TokenParameters tokenParameters,
+                Action<TokenResultErrorEventArgs> decodingErrorCallBack)
+        {
+            if (decodeParameters is null)
+            {
+                throw new ArgumentNullException(nameof(decodeParameters));
+            }
+
+            if (tokenParameters is null)
+            {
+                throw new ArgumentNullException(nameof(tokenParameters));
+            }
+
+            if (decodingErrorCallBack is null)
+            {
+                throw new ArgumentNullException(nameof(decodingErrorCallBack));
+            }
+
+            if (string.IsNullOrWhiteSpace(tokenParameters.Token))
+            {
+                throw new ArgumentNullException(nameof(tokenParameters.Token));
+            }
+
+            _decodingErrorCallBack = decodingErrorCallBack;
+
+            var tokenResult = new TokenResult();
+
+            try
+            {
+                IdentityModelEventSource.ShowPII = true;
+                var handler = new JwtSecurityTokenHandler();
+                JwtSecurityToken jwtSecurityToken = handler.ReadJwtToken(tokenParameters.Token);
+                tokenResult.Header = JsonHelper.Format(jwtSecurityToken.Header.SerializeToJson(), Indentation.TwoSpaces, false);
+                tokenResult.Payload = JsonHelper.Format(jwtSecurityToken.Payload.SerializeToJson(), Indentation.TwoSpaces, false);
+                tokenResult.TokenAlgorithm = tokenParameters.TokenAlgorithm;
+
+                if (decodeParameters.ValidateSignature)
+                {
+                    var signatureValid = ValidateTokenSignature(handler, decodeParameters, tokenParameters, tokenResult);
```

Comment:
Please follow Visual Studio warnings and use `bool` instead of `var` here. It improves the readability of the code.